src/redux/transactionsSlice.js

```diff
@@ -0,0 +1,52 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { listCustomerTransactions } from 'api';
+
+export const DEFAULT_TRANSACTIONS_NUMBER = 3;
+
+const initialState = {
+  transactionsList: [],
+  transactionsError: [],
+  showToggleButton: false,
+  isTransactionsSectionLoading: false,
+  isTransactionListExpanded: false
+};
+
+export const fetchListCustomerTransactions = createAsyncThunk(
+  'transactions/listCustomerTransactions',
+  async (_, { rejectWithValue }) => {
+    try {
+      const responseData = await listCustomerTransactions();
+      return responseData;
+    } catch (error) {
+      return rejectWithValue(error.message);
+    }
+  }
+);
+
+export const transactionsSlice = createSlice({
+  name: 'transactions',
+  initialState,
+  reducers: {
+    setIsTransactionListExpanded(state, { payload }) {
+      state.isTransactionListExpanded = payload;
+    }
```

Comment:
I see that we are using this function only to toggle the list, so we could also define it here, not in Transactions component. So more less it would be like:
```
toggleTransactionList(state, {payload}) {
 state. isTransactionListExpanded = !state.isTransactionListExpanded
} 
```
and then in Transactions component `toggleTransactionsList` could be replaced with this one